admin/bin/init-dev-storage.js

```diff
@@ -19,68 +19,69 @@ try {
 } catch (error) {
   console.error(error)
 }
-const azure = require('azure-storage')
-const bluebird = require('bluebird')
+const { TableClient } = require('@azure/data-tables')
+const { QueueServiceClient } = require('@azure/storage-queue')
 const names = require('../../deploy/storage/tables-queues.json')
 
 if (!process.env.AZURE_STORAGE_CONNECTION_STRING) {
   process.exitCode = -1
   console.error('env var $AZURE_STORAGE_CONNECTION_STRING is required')
 }
 
+const connectionString = process.env.AZURE_STORAGE_CONNECTION_STRING
+
 const queueNames = names.queues
 const tableNames = names.tables
 const poisonQueues = queueNames.map(q => q + '-poison')
 const allQueues = queueNames.concat(poisonQueues)
-const tableService = getPromisifiedService(azure.createTableService())
-const queueService = getPromisifiedService(azure.createQueueService())
+
+async function deleteTableEntries (tableName) {
+  const tableClient = TableClient.fromConnectionString(connectionString, tableName)
+  const entityIterator = tableClient.listEntities()
+  const deletions = []
+  for await (const entity of entityIterator) {
+    deletions.push(tableClient.deleteEntity(entity.partitionKey, entity.rowKey))
+  }
+  return Promise.all(deletions)
+}
 
 async function deleteTableEntities (tables) {
   const deletions = []
   for (let index = 0; index < tables.length; index++) {
     const table = tables[index]
-    const query = new azure.TableQuery() // Azure Table Storage has a max of 1000 records returned
-    let done = false
-    let batch = 1
-    while (!done) {
-      const data = await tableService.queryEntitiesAsync(table, query, null)
-      const entities = data.result.entries
-      if (entities.length === 0) {
-        done = true
-      }
-      console.log(`Found ${entities.length} entities to delete in batch ${batch++} from ${table}`)
-      entities.forEach(entity => {
-        deletions.push(tableService.deleteEntityAsync(table, entity))
-      })
-      await Promise.all(deletions)
-    }
+    deletions.push(deleteTableEntries(table))
   }
+  await Promise.all(deletions)
```

Comment:
I think `Promise.allSettled()` offers better functionality in this case, as deletions are not dependent on each other.  Available since node 12.9.0 https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled